src/App.tsx

```diff
@@ -1,14 +1,48 @@
-import React from 'react';
+import React, { useMemo, useState, useCallback } from 'react';
 import './App.scss';
+import debounce from 'lodash.debounce';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { People } from './components/People';
+
+const initialPeople: Person[] = peopleFromServer.map(person => ({
+  ...person,
+}));
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+
+  const applyQuery = useCallback(
+    debounce(setAppliedQuery, 1000),
+    [],
+  );
```

Comment:
Don't use the `useCallback` without a reason.
You can read more about this hook here: https://react.dev/reference/react/useCallback